### PR TITLE
(PDOC-88) Add additional spec and acceptance tests

### PR DESCRIPTION
**Pull request by whopper**

### CODE REVIEW

File: spec/unit/puppet-strings/yard/handlers/ruby/provider_handler_spec.rb [hunk 1]
```diff
@@ -27,6 +27,34 @@
     end
   end
 
+  describe 'parsing a provider with an invalid docstring assignment' do
+    let(:source) { <<-SOURCE
+Puppet::Type.type(:custom).provide :linux do
+  @doc = 123
```
Comment (er0ck) [inline spec/unit/puppet-strings/yard/handlers/ruby/provider_handler_spec.rb]: curiosity to me.. is this what you referenced for puppet 3x?  do the various types of docstrings have to be instance vars?

Comment (whopper) [inline spec/unit/puppet-strings/yard/handlers/ruby/provider_handler_spec.rb]: @er0ck I think you *can* reference the @doc instance var, but the only record of this being a common thing I could find was https://projects.puppetlabs.com/issues/11404.

I added the test to cover this bit of strings code: https://github.com/puppetlabs/puppet-strings/blob/master/lib/puppet-strings/yard/handlers/ruby/provider_handler.rb#L52-L55

File: spec/acceptance/running_strings_generate.rb [hunk 2]
```diff
@@ -18,6 +18,10 @@ def read_file_on(host, filename)
     expect(read_file_on(master, '/root/doc/puppet_classes/test.html')).to include('Class: test')
   end
 
+  it 'should generate documentation for puppet functions' do
+    expect(read_file_on(master, '/root/doc/puppet_functions_puppet/strings_3A_3Aadd.html')).to include('Adds two integers together.')
```
Comment (er0ck) [inline spec/acceptance/running_strings_generate.rb]: should this also test for the `@params` and `@return` and `@example`?

Comment (whopper) [inline spec/acceptance/running_strings_generate.rb]: @er0ck yeah, it may as well. Updated to check for those.

**Comment by whopper** [comment]
The acceptance test may not be working correctly - looking into it.

**Comment by whopper** [comment]
Added two more tests for types and providers - all pass:
```
Generating module documentation using generate action
  should generate documentation for manifests
  should generate documentation for puppet functions
  should generate documentation for 3x functions
  should generate documentation for 4x functions
  should generate documentation for custom types
  should generate documentation for custom providers
```

**Pull request merged**